src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts

```diff
@@ -51,129 +45,65 @@ export class MLSConversationVerificationStateHandler {
+    conversation.mlsVerificationState(state);
+    this.onConversationVerificationStateChange({
+      conversationEntity: conversation,
+      conversationVerificationState: state,
+      verificationMessageType: VerificationMessageType.UNVERIFIED,
     });
-    if (conversationVerificationState) {
-      this.onConversationVerificationStateChange({
-        conversationEntity,
-        conversationVerificationState,
-        verificationMessageType: VerificationMessageType.UNVERIFIED,
-        userIds,
-      });
-    }
-  };
+  }
 
   /**
    * This function checks if the conversation is degraded and if it is, it will verify it
-   * @param conversationEntity
+   * @param conversation
    * @param userIds
    */
-  private verifyConversation = async (conversationEntity: MLSConversation) => {
-    this.logger.log(`Conversation ${conversationEntity.name} will be verified`);
-
-    const conversationVerificationState = attemptChangeToVerified({conversationEntity, logger: this.logger});
-
-    if (conversationVerificationState) {
-      this.onConversationVerificationStateChange({
-        conversationEntity,
-        conversationVerificationState,
-      });
-    }
-  };
-
-  /**
-   * This function returns the WireIdentity of all userDeviceEntities in a conversation, as long as they have a certificate.
-   * If the conversation has userDeviceEntities without a certificate, it will not be included in the returned array
-   *
-   * It also updates the isMLSVerified observable of all the devices in the conversation
-   */
-  private updateUserDevices = async (conversation: MLSConversation) => {
-    const userEntities = conversation.getAllUserEntities();
-    const allClients: ClientEntity[] = [];
-    const allIdentities: WireIdentity[] = [];
-    userEntities.forEach(async userEntity => {
-      let devices = userEntity.devices();
-      // Add the localClient to the devices array if it is the current user
-      if (userEntity.isMe && userEntity.localClient) {
-        devices = [...devices, userEntity.localClient];
-      }
-      const deviceUserPairs = devices
-        .map(device => ({
-          [device.id]: userEntity.qualifiedId,
-        }))
-        .reduce((acc, current) => {
-          return {...acc, ...current};
-        }, {});
-      const identities = await this.core.service!.e2eIdentity!.getUserDeviceEntities(
-        conversation.groupId,
-        deviceUserPairs,
-      );
-      identities.forEach(async identity => {
-        const verified = await this.isCertificateActiveAndValid(identity.certificate);
-        if (verified) {
-          const device = devices.find(device => device.id === identity.clientId);
-          /**
-           * ToDo: Change the current implementation of isMLSVerified to be stored in Zustand instead of ko.observable
-           */
-          device?.meta.isMLSVerified?.(true);
-          allIdentities.push(identity);
-        }
-      });
-      allClients.push(...devices);
```

Comment:
This was not the correct way to compute device verification. Instead CoreCrypto should be queried to know if a device is verified or not. 

The problem with this method is that local devices list is not necessarily up to date (it's just a cache). So we will miss a lot of devices with this method